internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -335,49 +335,126 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 				}
-			}
 
-			if (customerEncryptionEnabled.(bool) || defaultStorageFirewallEnabled.(bool) || infrastructureEncryptionEnabled.(bool) || managedServicesCMK.(string) != "" || managedDiskCMK.(string) != "") && !strings.EqualFold("premium", newSku.(string)) {
-				return fmt.Errorf("'customer_managed_key_enabled', 'default_storage_firewall_enabled', 'infrastructure_encryption_enabled', 'managed_disk_cmk_key_vault_key_id' and 'managed_services_cmk_key_vault_key_id' are only available with a 'premium' workspace 'sku', got %q", newSku)
-			}
+				return nil
+			}),
 
-			return nil
-		}),
+			// Once compliance security profile has been enabled, disabling it will force a workspace replacement
+			pluginsdk.ForceNewIfChange("enhanced_security_compliance.0.compliance_security_profile_enabled", func(ctx context.Context, old, new, meta interface{}) bool {
+				return old.(bool) && !new.(bool)
+			}),
+
+			// Once a compliance standard is enabled, disabling it will force a workspace replacement
+			pluginsdk.ForceNewIfChange("enhanced_security_compliance.0.compliance_security_profile_standards", func(ctx context.Context, old, new, meta interface{}) bool {
+				removedStandards := old.(*pluginsdk.Set).Difference(new.(*pluginsdk.Set))
+				return removedStandards.Len() > 0
+			}),
+
+			// Compliance security profile requires automatic cluster update and enhanced security monitoring to be enabled
+			pluginsdk.CustomizeDiffShim(func(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+				_, complianceSecurityProfileEnabled := d.GetChange("enhanced_security_compliance.0.compliance_security_profile_enabled")
+				_, automaticClusterUpdateEnabled := d.GetChange("enhanced_security_compliance.0.automatic_cluster_update_enabled")
+				_, enhancedSecurityMonitoringEnabled := d.GetChange("enhanced_security_compliance.0.enhanced_security_monitoring_enabled")
+
+				if complianceSecurityProfileEnabled.(bool) && (!automaticClusterUpdateEnabled.(bool) || !enhancedSecurityMonitoringEnabled.(bool)) {
+					return fmt.Errorf("'automatic_cluster_update_enabled' and 'enhanced_security_monitoring_enabled' must be set to true when using 'compliance_security_profile'")
+				}
+
+				return nil
+			}),
+
+			// compliance standards cannot be specified without enabling compliance profile
+			pluginsdk.CustomizeDiffShim(func(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+				_, complianceSecurityProfileEnabled := d.GetChange("enhanced_security_compliance.0.compliance_security_profile_enabled")
+				_, complianceStandards := d.GetChange("enhanced_security_compliance.0.compliance_security_profile_standards")
+
+				if !complianceSecurityProfileEnabled.(bool) && complianceStandards.(*pluginsdk.Set).Len() > 0 {
+					return fmt.Errorf("'compliance_security_profile_standards' cannot be set when 'compliance_security_profile_enabled' is false")
+				}
+
+				return nil
+			}),
```

Comment:
There is an opportunity to simplify the schema and Customize Diff behaviour of the resource if this is the case, i.e. if `compliance_security_profile_standards` has been set then we enable compliance security profile for the user.

One could argue that since these are security and compliance related settings that it would be better to be explicit, but I can't see why a user would want to set a value for `compliance_security_profile_standards` and not want the compliance security profile enabled.

Reply:
> i.e. if compliance_security_profile_standards has been set then we enable compliance security profile for the user

This won't work, as it is possible to enable compliance security profile with "NONE" standard.

The screencast below will give an idea of the behaviour of the 4 inputs in the UI.

[screencast link](
https://microsoftapc-my.sharepoint.com/:f:/g/personal/tangerry_microsoft_com/EjeTUHpKS69Dg03AJaUWUgYBkqTAlXrG861kiYdg8XmvjA?email=steph%40hashicorp.com&e=20bDpb)

I think modelling them as 4 independent inputs that gets validated is the simplest way, otherwise we have to implement a complex implicit imply rule.

Reply:
Ah sorry, I've reuploaded the screencast in a different hosting: [screencast link (gdrive)](https://drive.google.com/file/d/1Bwnx8D4JmrJbXjHio0jBPHGBQ7ga1yoo/view?usp=drive_link)